web/js/codeworld_shared.js

```diff
@@ -482,46 +497,52 @@ function registerStandardHints(successFunc) {
     // 3. Others, to be presented as fuzzy matches.
     const found = [[], [], []];
 
-    const hints = Object.keys(window.codeWorldSymbols);
-    for (let i = 0; i < hints.length; i++) {
-      const hint = hints[i];
-      const parts = hint.split(/\.(?=[^.]+$)/);
+    for (const [hintName, hintProps] of Object.entries(
+      window.codeWorldSymbols
+    )) {
+      const parts = hintName.split(/\.(?=[^.]+$)/);
       const hintPrefix = parts.length < 2 ? '' : `${parts[0]}.`;
-      const hintIdent = parts.length < 2 ? hint : parts[1];
-      if (!VAR_OR_CON.test(hintIdent)) continue;
-      if (window.codeWorldSymbols[hint].module) {
-        if (hint.startsWith(prefix)) {
+      const hintIdent = parts.length < 2 ? hintName : parts[1];
+
+      if (!VAR_OR_CON.test(hintIdent)) {
+        continue;
+      }
+
+      if (hintProps.module) {
+        if (hintName.startsWith(prefix)) {
           const candidate = {
-            text: window.codeWorldSymbols[hint].insertText.substr(
-              prefix.length
-            ),
-            details: window.codeWorldSymbols[hint],
+            text: hintProps.insertText.substr(prefix.length),
+            details: hintProps,
             render: (elem) => {
-              renderDeclaration(elem, window.codeWorldSymbols[hint], 50);
+              renderDeclaration(elem, hintProps, 50);
             },
           };
-          if (hint === prefix + token.string) {
+          if (hintName === prefix + token.string) {
             found[0].push(candidate);
-          } else if (hint.startsWith(prefix + term)) {
+          } else if (hintName.startsWith(prefix + term)) {
             found[1].push(candidate);
           } else {
-            found[2].push(candidate);
+            if (hintProps.definingModule === replacementTerms.definingModule) {
+              found[2].push(candidate);
+            }
           }
         }
       } else if (hintPrefix === prefix) {
         const candidate = {
-          text: window.codeWorldSymbols[hint].insertText,
-          details: window.codeWorldSymbols[hint],
+          text: hintProps.insertText,
+          details: hintProps,
           render: (elem) => {
-            renderDeclaration(elem, window.codeWorldSymbols[hint], 50);
+            renderDeclaration(elem, hintProps, 50);
           },
         };
         if (hintIdent === token.string) {
           found[0].push(candidate);
         } else if (hintIdent.startsWith(term)) {
           found[1].push(candidate);
         } else {
-          found[2].push(candidate);
+          if (hintProps.definingModule === replacementTerms.definingModule) {
```

Comment:
I don't think this is right, either.  You still want to add the candidate under its original name, even if it isn't in the right module for replacement terms to apply.

Reply:
Okay, I'll remove this condition. Thanks for the explanation.